utils/frame/remote-externalities/src/lib.rs

```diff
@@ -320,72 +382,147 @@ where
 		Ok(keys)
 	}
 
-	/// Synonym of `rpc_get_pairs_unsafe` that uses paged queries to first get the keys, and then
+	/// Synonym of `getPairs` that uses paged queries to first get the keys, and then
 	/// map them to values one by one.
 	///
 	/// This can work with public nodes. But, expect it to be darn slow.
 	pub(crate) async fn rpc_get_pairs_paged(
 		&self,
 		prefix: StorageKey,
 		at: B::Hash,
+		pending_ext: &mut TestExternalities,
 	) -> Result<Vec<KeyValue>, &'static str> {
-		let keys = self.rpc_get_keys_paged(prefix, at).await?;
-		let keys_count = keys.len();
-		log::debug!(target: LOG_TARGET, "Querying a total of {} keys", keys.len());
-
-		let mut key_values: Vec<KeyValue> = vec![];
-		let client = self.as_online().rpc_client();
-		for chunk_keys in keys.chunks(BATCH_SIZE) {
-			let batch = chunk_keys
-				.iter()
-				.cloned()
-				.map(|key| ("state_getStorage", rpc_params![key, at]))
-				.collect::<Vec<_>>();
+		let keys =
+			time_block!("rpc_get_keys_paged", { self.rpc_get_keys_paged(prefix, at).await? });
+		let client = self.as_online().transport.remote_client.clone().unwrap();
+		let thread_chunk_size =
+			((keys.len() + self.as_online().threads - 1) / self.as_online().threads).max(1);
```

Comment:
suggestion: just to avoid strange failure - `saturating_div`?

Reply:
what I actually need is a simple `div_ceil`, not sure how `saturating_div` helps with that?